ts/src/config.ts

```diff
@@ -61,25 +61,35 @@ export interface Config extends AuthenticationConfig {
   disableHeap?: boolean;
 }
 
-/**
- * Subtype of Config that declares some properties to be non-optional. This
- * helps avoid null checks in the profiler code.
- */
-export interface ProfilerConfig extends Config {
+// Interface for an initialized config.
+export interface ProfilerConfig extends AuthenticationConfig {
+  projectId?: string;
   logLevel: number;
-  serviceContext: {service?: string; version?: string;};
+  serviceContext: {service: string; version?: string;};
   instance: string;
   zone: string;
   disableTime: boolean;
   disableHeap: boolean;
+  timeIntervalMicros: number;
+  heapIntervalBytes: number;
+  heapMaxStackDepth: number;
+  backoffMillis: number;
 }
 
 // Default values for configuration for a profiler.
-export const defaultConfig: ProfilerConfig = {
+export const defaultConfig = {
   logLevel: 1,
   serviceContext: {},
   disableHeap: false,
   disableTime: false,
   instance: '',
   zone: ''
 };
+
+// Configuration parameters set internally, not by users.
+export const internalConfig = {
+  timeIntervalMicros: 1000,
+  heapIntervalBytes: 512 * 1024,
+  heapMaxStackDepth: 64,
+  backoffMillis: 1000,  // 1 second
```

Comment:
One second backoff without exponential increase is way too low.

Reply:
Done -- increasing backoff to 5 minutes.